client/src/components/ShelfCreator.js

```diff
@@ -0,0 +1,341 @@
+import styled from 'styled-components/macro';
+import { useState } from 'react';
+
+import getShelfBorders from '../lib/shelfBorders';
+
+export default function ShelfCreator() {
+  const initialShelf = {
+    name: '',
+    columns: [],
+    color: '',
+  };
+
+  const [shelf, setShelf] = useState(initialShelf);
+
+  function updateShelf(event) {
+    const fieldName = event.target.name;
+    let fieldValue = event.target.value;
+    if (event.target.id === 'columns') {
+      fieldValue = [];
+      for (let i = 0; i < event.target.value; i++) {
+        fieldValue.push({
+          column: i + 1,
+          compartments: [],
+          width: 1,
+          height: 1,
+        });
+      }
+    }
+
+    setShelf({ ...shelf, [fieldName]: fieldValue });
+  }
+
+  function updateColumn(event, index) {
+    const fieldName = event.target.name;
+    let fieldValue = Number(event.target.value);
+    const newShelf = { ...shelf };
+    if (event.target.id === 'compartments') {
+      fieldValue = [];
+      for (let i = 0; i < event.target.value; i++) {
+        fieldValue.push({ compartment: i + 1 });
+      }
+    }
+    newShelf.columns[index] = {
+      ...newShelf.columns[index],
+      [fieldName]: fieldValue,
+    };
+    setShelf(newShelf);
+  }
+
+  function shelfWidth(index) {
+    const columnWidth =
+      shelf.columns[index].width === undefined
+        ? 100 / shelf.columns.length
+        : (100 / shelf.columns.length) * shelf.columns[index].width;
+    return columnWidth;
+  }
+
+  function checkForHeight(value) {
+    return shelf.columns.some((column) => column.height === value);
+  }
+
+  function shelfHeight(index) {
+    let columnHeight = () => {
+      if (checkForHeight(3)) return (100 / 3) * shelf.columns[index].height;
+      if (checkForHeight(2)) return (100 / 2) * shelf.columns[index].height;
+      else return (columnHeight = 100);
+    };
```

Comment:
Is there a reason why a function is returned instead of the columnHeight´s value?